platform/darwin/src/MGLCircleStyleLayer.mm

```diff
@@ -75,79 +90,121 @@ - (void)removeFromMapView:(MGLMapView *)mapView
     auto mbglValue = MGLStyleValueTransformer<float, NSNumber *>().toPropertyValue(circleOpacity);
-    self.rawLayer->setCircleOpacity(mbglValue);
+    _rawLayer->setCircleOpacity(mbglValue);
 }
 
 - (MGLStyleValue<NSNumber *> *)circleOpacity {
-    auto propertyValue = self.rawLayer->getCircleOpacity() ?: self.rawLayer->getDefaultCircleOpacity();
+    auto propertyValue = _rawLayer->getCircleOpacity() ?: _rawLayer->getDefaultCircleOpacity();
     return MGLStyleValueTransformer<float, NSNumber *>().toStyleValue(propertyValue);
 }
 
 - (void)setCircleTranslate:(MGLStyleValue<NSValue *> *)circleTranslate {
     auto mbglValue = MGLStyleValueTransformer<std::array<float, 2>, NSValue *>().toPropertyValue(circleTranslate);
-    self.rawLayer->setCircleTranslate(mbglValue);
+    _rawLayer->setCircleTranslate(mbglValue);
 }
 
 - (MGLStyleValue<NSValue *> *)circleTranslate {
-    auto propertyValue = self.rawLayer->getCircleTranslate() ?: self.rawLayer->getDefaultCircleTranslate();
+    auto propertyValue = _rawLayer->getCircleTranslate() ?: _rawLayer->getDefaultCircleTranslate();
     return MGLStyleValueTransformer<std::array<float, 2>, NSValue *>().toStyleValue(propertyValue);
 }
 
 - (void)setCircleTranslateAnchor:(MGLStyleValue<NSValue *> *)circleTranslateAnchor {
-    auto mbglValue = MGLStyleValueTransformer<mbgl::style::TranslateAnchorType, NSValue *>().toPropertyValue(circleTranslateAnchor);
-    self.rawLayer->setCircleTranslateAnchor(mbglValue);
+    if ([circleTranslateAnchor isKindOfClass:[MGLStyleFunction class]]) {
+        MGLStyleFunction<NSValue *> *function = (MGLStyleFunction<NSValue *> *)circleTranslateAnchor;
+        __block std::vector<std::pair<float, mbgl::style::TranslateAnchorType>> mbglStops;
+        [function.stops enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull zoomKey, MGLStyleValue<NSValue *> * _Nonnull stopValue, BOOL * _Nonnull stop) {
+            id value = [(MGLStyleConstantValue<NSValue *> *)stopValue rawValue];
+            MGLCircleTranslateAnchor circleTranslateAnchorValue;
+            [value getValue:&circleTranslateAnchorValue];
+            auto str = mbgl::Enum<MGLCircleTranslateAnchor>::toString(circleTranslateAnchorValue);
+            auto mbglValue = mbgl::Enum<mbgl::style::TranslateAnchorType>::toEnum(str).value_or(_rawLayer->getDefaultCircleTranslateAnchor().asConstant());
+            auto mbglStopValue = mbgl::style::PropertyValue<mbgl::style::TranslateAnchorType>(mbglValue);
+            mbglStops.emplace_back(zoomKey.floatValue, mbglStopValue.asConstant());
+        }];
+        auto func = mbgl::style::Function<mbgl::style::TranslateAnchorType>({{mbglStops}}, function.base);
+        _rawLayer->setCircleTranslateAnchor(func);
+        return;
+    }
+    id value = [(MGLStyleConstantValue<NSValue *> *)circleTranslateAnchor rawValue];
+    MGLCircleTranslateAnchor circleTranslateAnchorValue;
+    [value getValue:&circleTranslateAnchorValue];
```

Comment:
In the previous code, if `circleTranslateAnchor` was `nil`, we’d [reset](https://github.com/mapbox/mapbox-gl-native/blob/307f8e4f41a440fdfcd9c7bb1833110ab9278ccf/platform/darwin/src/MGLStyleValue_Private.h#L52) the value in mbgl to an implicit default value. In this code, if `circleTranslateAnchor` is `nil`, `circleTranslateAnchorValue` will be `MGLCircleTranslateAnchorMap`, even if the implicit default would’ve been `mbgl::style::TranslateAnchorType::Viewport`.